app/subapps/browser/views/charm.js

```diff
@@ -354,18 +353,15 @@ YUI.add('subapp-browser-charmview', function(Y) {
         this._renderCharmView(this.get('entity'));
         this.hideIndicator(this.get('renderTo'));
       } else {
-        this.get('store').charm(this.get('entityId'), {
-          'success': function(data) {
-            var charm = new models.Charm(data.charm);
-            if (data.metadata) {
-              charm.set('metadata', data.metadata);
-            }
-            this.set('charm', charm);
-            this._renderCharmView(charm);
-            this.hideIndicator(this.get('renderTo'));
-          },
-          'failure': this.apiFailure
-        }, this);
+        this.get('charmstore').getEntity(
+            this.get('entityId'),
+            function(charms) {
+              var charm = charms[0];
```

Comment:
The data returned from getEntity goes through a few transform methods, one of them handles transforming objects into the Charm models. Because this method is used in many different places it returns an array of items. In this case though we know for sure that we are only going to get a single response back from the api so that's why I'm grabbing the first one.